tests/common/helpers/parallel_utils.py

```diff
@@ -1,5 +1,180 @@
+import fcntl
+import logging
+import multiprocessing
+from datetime import datetime
+from enum import Enum
+from typing import Tuple
+
 from tests.common import config_reload
+from tests.common.helpers.assertions import pytest_assert as pt_assert
+from tests.common.utilities import wait_until
+
+logger = logging.getLogger(__name__)
 
 
 def config_reload_parallel_compatible(node, results, *args, **kwargs):
     return config_reload(node, *args, **kwargs)
+
+
+def is_initial_checks_active(request):
+    parallel_state_file = request.config.getoption("--parallel_state_file")
+    try:
+        current_parallel_status = read_last_line_from_file(parallel_state_file).split(',')[1]
+        logger.debug("current parallel status: {}".format(current_parallel_status))
+        return current_parallel_status.endswith("started")
+    except FileNotFoundError:
+        logger.info("File {} not found".format(parallel_state_file))
+        return False
+
+
+def read_last_line_from_file(file):
+    with open(file, 'r') as f:
+        fcntl.flock(f, fcntl.LOCK_EX)
+        try:
+            return f.readlines()[-1].strip()
```

Comment:
also for this one i'm not sure how big the file will be but perhaps it will be more efficient if we just extract the last line out instead of using readlines() since readlines() will create an array to store the lines.

Reply:
The file size is related to the number of DUTs in the testbed. Given that we don't have a testbed that contains large number of DUTs, the file size will not be very big, but this is still a very good suggestion to improve the efficiency. Let me see what I can do.

Reply:
Maybe we can put this as a TODO for future improvements?

Reply:
@cyw233  I think it should be fine. Maybe we can take a look at this if we experience bad performance. But from our discussion having 4-5 lines in a file I don't think it will affect much.